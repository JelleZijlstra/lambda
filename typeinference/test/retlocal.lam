let f = \x.
	data option a = None | Some a in
	Some x in

let g = \x.
	data option a b = None | Some a b in
	Some x true in
(f, g)
