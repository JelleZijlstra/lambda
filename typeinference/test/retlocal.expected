%%stdout%%
Result: module
	let __result__ = (\x. type option a = None  | Some 'a in Some x, \x. type option a b = None  | Some 'b 'a in Some x true)
end
%%stderr%%
